sdk/identity/azure-identity/azure/identity/_credentials/azure_powershell.py

```diff
@@ -34,9 +34,20 @@
     exit
 }}
 
-$token = Get-AzAccessToken -ResourceUrl '{}'{}
+$useSecureString = $m.Version -ge [version]"2.17.0"
 
-Write-Output "`nazsdk%$($token.Token)%$($token.ExpiresOn.ToUnixTimeSeconds())`n"
+$command = "Get-AzAccessToken -ResourceUrl '{}'{}"
+
+if ($useSecureString) {{
+    $command += " -AsSecureString"
+}}
+
```

Comment:
Do we need try catch here?

e.g. 
try {{
    $token = Invoke-Expression $command
    $tokenValue = $token.Token
    if ($useSecureString) {{
        $tokenValue = $tokenValue | ConvertFrom-SecureString -AsPlainText
    }}
    Write-Output "`nazsdk%$($tokenValue)%$($token.ExpiresOn.ToUnixTimeSeconds())`n"
}} catch {{
    Write-Output "ERROR: Failed to retrieve token"
    exit 1
}}

Reply:
Don't think it is necessary, since if any of the commands here fail, the script will exit with a non-zero exit code anyway.  Our [`raise_for_error` method](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/azure_power) will log the stderr as debug output and a `CredentialUnavailableError` will be raised.

Reply:
I am thinking if we can expose more error details here and print it out in debug logging.

Reply:
After testing, if the cmdlets, like Get-AzAccessToken, fail for whatever reason, the error is already exposed in stderr which we will print to the debug log. So I don't think we need to do any additional steps. In any case, I am learning I should avoid Invoke-Expression, so I am going to refactor this a bit.